app/topics_handler.go

```diff
@@ -41,12 +43,48 @@ func (th *TopicsHandler) calculatePreviousBlockApproxTime(ctx sdk.Context, block
 	return previousBlockApproxTime, nil
 }
 
+func sortAndSelectTopNReputerNonces(reputerRequestNonces *emissionstypes.ReputerRequestNonces, N int) []*emissionstypes.ReputerRequestNonce {
+	// Define a custom sorting function
+	sorter := func(i, j int) bool {
+		return reputerRequestNonces.Nonces[i].ReputerNonce.BlockHeight > reputerRequestNonces.Nonces[j].ReputerNonce.BlockHeight
+	}
+
+	// Sort the nonces slice using the custom sorting function
+	sort.Slice(reputerRequestNonces.Nonces, sorter)
+
+	// Select the top N latest elements
+	var topN []*emissionstypes.ReputerRequestNonce
+	if len(reputerRequestNonces.Nonces) <= N {
+		topN = reputerRequestNonces.Nonces
+	} else {
+		topN = reputerRequestNonces.Nonces[:N]
+	}
+	return topN
+}
+
+func sortAndSelectTopNWorkerNonces(workerNonces emissionstypes.Nonces, N int) []*emissionstypes.Nonce {
```

Comment:
maybe we don't need to sort? per our convo

Reply:
changes done